.github/workflows/test_openvino.yml

```diff
@@ -20,40 +18,55 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.12"]
+        test-pattern:
+          [
+            "*modeling*",
+            "*diffusion*",
+            "*quantization*",
+            "*training*",
+            "*export*",
+          ]
         transformers-version: ["4.36.0", "latest"]
-        os: [ubuntu-latest]
 
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-22.04
+
     steps:
-      - uses: actions/checkout@v4
-      - name: Setup Python ${{ matrix.python-version }}
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Python
         uses: actions/setup-python@v5
         with:
-          python-version: ${{ matrix.python-version }}
-
-      - name: Install lowest compatible transformers version
-        if: ${{ matrix.transformers-version != 'latest' }}
-        run: pip install transformers==${{ matrix.transformers-version }} accelerate==0.*
+          python-version: 3.9
 
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          # install PyTorch CPU version to avoid installing CUDA packages on GitHub runner without GPU
+          pip install --upgrade pip
           pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu
-          pip install .[openvino,openvino-tokenizers,tests,diffusers] onnxruntime
+          pip install .[openvino,openvino-tokenizers,diffusers,tests] transformers[testing]
+
+      - if: ${{ matrix.transformers-version != 'latest' }}
+        name: Downgrade Transformers and Accelerate
+        run: |
+          pip install transformers==${{ matrix.transformers-version }} accelerate==0.*
+
+      - if: ${{ matrix.test-pattern == '*modeling*' }}
+        name: Uninstall NNCF
+        run: |
+          pip uninstall -y nncf
```

Comment:
Any reason we uninstall NNCF here? We can help to resolve if there are any issues caused by the NNCF presence.

Reply:
No issue, just to make sure NNCF is not a dep for OV modeling (I guess this was the case at some point)
Also https://github.com/huggingface/optimum-intel/pull/975#discussion_r1827733818